pennylane/decomposition/decomposition_graph.py

```diff
@@ -196,7 +211,10 @@ def _recursively_add_op_node(self, op_node: CompressedResourceOp) -> int:
         self._all_op_indices[op_node] = op_node_idx
 
         if op_node.name in self._gate_set:
-            self._graph.add_edge(self._start, op_node_idx, 1)
+            self._graph.add_edge(
+                self._start, op_node_idx,
+                self._weights[op_node.name],
+            )
```

Comment:
```suggestion
            self._graph.add_edge(self._start, op_node_idx, self._weights[op_node.name])
```
When using `black`, use `black -l 100 ...`